### PR TITLE
chore(tests): move opentracing to riot

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Moves opentracing tests from tox to riot. Pins problematic gevent versions (v1.3, v1.4) in python>3.7 tests

## Checklist
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -422,8 +422,8 @@ jobs:
     <<: *contrib_job
     parallelism: 8
```
Comment (Kyle-Verhoog) [inline .circleci/config.yml:423]: can we drop the parallelism now? probably don't need it if running with riot

Comment (brettlangdon) [reply .circleci/config.yml:423]: do we even have 8 Python runtimes for this job?

Comment (mabdinur) [inline .circleci/config.yml:423]:
```suggestion
```

**Review by Kyle-Verhoog — APPROVED** [review]

File: riotfile.py [hunk 2]
```diff
@@ -2296,6 +2296,78 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                                "gevent": ["~=1.1.0", "~=1.2.0", "~=1.3.0"],
+                                "greenlet": "~=1.0",
+                            },
+                        ),
+                        Venv(
+                            pys=select_pys(min_version="3.5", max_version="3.6"),
+                            pkgs={
+                                "gevent": ["~=1.1.0", "~=1.2.0", "~=1.3.0"],
+                                "greenlet": "~=1.0",
+                            },
```
Comment (Yun-Kim) [inline riotfile.py:2336]: Is there a reason these were separated? Looks like we can run the same dependencies in one venv.

File: riotfile.py [hunk 3]
```diff
@@ -2296,6 +2296,78 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                        ),
```
Comment (mabdinur) [inline riotfile.py:2337]:
```suggestion
                        Venv(
                            pys=select_pys(max_version="3.6"),
                            pkgs={
                                "gevent": ["~=1.1.0", "~=1.2.0],
                                "greenlet": "~=1.0",
                            },
                        ),
```

File: riotfile.py [hunk 4]
```diff
@@ -2296,6 +2296,71 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                        "tornado": ["~=4.4.0", "~=4.5.0", "~=5.0.0", "~=5.1.0"],
+                    },
+                ),
+                Venv(
+                    command="pytest {cmdargs} tests/opentracer/test_tracer_gevent.py",
+                    venvs=[
+                        Venv(
+                            pys=select_pys(max_version="3.6"),
+                            pkgs={
+                                "gevent": ["~=1.1.0", "~=1.2.0],
```
Comment (mabdinur) [inline riotfile.py:2327]:
```suggestion
                                "gevent": ["~=1.1.0", "~=1.2.0"],
```

**Review by Yun-Kim — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**